app/src/main/java/com/github/geohunt/app/maps/GoogleMapDisplay.kt

```diff
@@ -1,86 +1,138 @@
+import com.google.maps.android.compose.widgets.ScaleBar
 import java.time.LocalDateTime
 import java.time.Month
 import kotlin.math.PI
 import kotlin.math.cos
 import kotlin.math.pow
 
-private val epflCoordinates = LatLng(46.51958, 6.56398)
-
 /*
  * Displays the Google Map and its content
  *
  * @param modifier the modifier
  * @param cameraPosition the camera position
  * @param viewModel the view model used to access challenges from the database
- * @param content the content
  */
 @Composable
 fun GoogleMapDisplay(
     modifier: Modifier = Modifier,
-    cameraPosition: CameraPosition = CameraPosition(epflCoordinates, 10f, 0f, 0f),
     viewModel: MapsViewModel = viewModel(factory = MapsViewModel.Factory),
-    content: @Composable () -> Unit = {}
+    setCameraPosition: CameraPosition? = null,
 ) {
     val uiSettings by remember { mutableStateOf(MapUiSettings(compassEnabled = false)) }
-    val mapProperties by remember { mutableStateOf(MapProperties(mapType = MapType.NORMAL)) }
+    val mapProperties by remember { mutableStateOf(MapProperties(
+        mapType = MapType.NORMAL,
+        isMyLocationEnabled = true,
+        isBuildingEnabled = true,
+        isIndoorEnabled = true,
+    )) }
     val mapVisible by remember { mutableStateOf(true) }
-    val cameraPositionState = rememberCameraPositionState {
-        position = cameraPosition
+
+    val loc = remember { mutableStateOf<Location?>(null) }
+    val curLoc = viewModel.currentLocation.collectAsState()
+
+    DisposableEffect(viewModel) {
+        onDispose {
+            viewModel.reset()
+        }
     }
 
-    if (mapVisible) {
-        GoogleMap(
-            modifier = modifier,
-            cameraPositionState = cameraPositionState,
-            properties = mapProperties,
-            uiSettings = uiSettings,
-        ) {
-            val coordinateCenter = cameraPositionState.position.target
-            val zoom = cameraPositionState.position.zoom
-            val latitude = coordinateCenter.latitude
-            val longitude = coordinateCenter.longitude
-
-            val radius = 38000 / 2.0.pow((zoom - 3).toDouble()) * cos(latitude * PI / 180)
-            val location = Location(latitude, longitude)
-            val neighboringSectors = location.getNeighboringSectors(radius)
-
-            val markers = remember { mutableStateListOf<Marker>() }
-            viewModel.updateFetchableChallenges(neighboringSectors)
-            val challenges = viewModel.challenges.collectAsStateWithLifecycle()
-            challenges.value?.forEach {
-                val marker = Marker(
-                    title = it.id,
-                    image = it.photoUrl,
-                    coordinates = LatLng(it.location.latitude, it.location.longitude),
-                    expiryDate = it.expirationDate ?: LocalDateTime.of(2024, Month.MAY, 1, 19, 0),
-                )
-                markers.add(marker)
-            }
+    RequireFineLocationPermissions {
+    }
```

Comment:
Why do you think `RequireFineLocationPermissions` takes a callback

Reply:
But I can still use this function as it is right now?

Reply:
Yeah but you are not considering what happen if the user does not grant your permission

Reply:
In which case, in your view model when your do startLocationUpdate or idk how it's called it will either crash or do nothing

Reply:
So please fix this issue

Reply:
I have updated and tested my code so that it now handles correctly the case when the user does not grant the application the permissions to access to the device's location.